Lib/test/support/__init__.py

```diff
@@ -1900,21 +1899,54 @@ def _run_suite(suite):
         raise TestFailed(err)
 
 
-def _match_test(test):
-    global match_tests
+# By default, don't filter tests
+_match_test_func = None
 
-    if match_tests is None:
+
+def match_test(test):
+    # Function used by support.run_unittest() and regrtest --list-cases
+    if _match_test_func is None:
         return True
-    test_id = test.id()
+    else:
+        return _match_test_func(test.id())
+
+
+def _is_full_match_test(pattern):
+    # If a pattern contains at least one dot, it's considered as a full test
+    # identifier like 'test.test_os.FileTests.test_access'.
+    #
+    # Reject patterns which contain fnmatch patterns: '*', '?', '[...]'
+    # or '[!...]'. For example, reject 'test_access*'.
+    return ('.' in pattern) and all(char not in '?*[]' for char in pattern)
 
-    for match_test in match_tests:
-        if fnmatch.fnmatchcase(test_id, match_test):
-            return True
 
-        for name in test_id.split("."):
-            if fnmatch.fnmatchcase(name, match_test):
+def set_match_tests(patterns):
+    global _match_test_func
+
+    if not patterns:
+        _match_test_func = None
+    elif all(_is_full_match_test(pattern) for pattern in patterns):
```

Comment:
Could be `all(map(_is_full_match_test, patterns))`.

Reply:
Right, but I prefer list comprehensions. IMHO it's more readable.

Reply:
`map()` is already used twice few lines below.

Reply:
Oh right, that's the copied I copied from your PR :-D You won, I updated my PR to address your latest review.